instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/RuleParser.java

```diff
@@ -40,9 +46,103 @@ public static RuleParser get() {
 
   private RuleParser() {}
 
-  public JmxConfig loadConfig(InputStream is) throws Exception {
-    Yaml yaml = new Yaml(new Constructor(JmxConfig.class));
-    return yaml.load(is);
+  @SuppressWarnings("unchecked")
+  public JmxConfig loadConfig(InputStream is) {
```

Comment:
I think that the whole JMX config parsing module needs some further refactoring -- we could simplify the code, remove mutability, and clearly separate parsing from converting the config to its "executable" form. Not in this PR though, let's not mix concerns.